tools/analyzer_plugin/test/integration/assists/toggle_statefulness_test.dart

```diff
@@ -0,0 +1,224 @@
+import 'dart:async';
+
+import 'package:analyzer_plugin/utilities/assist/assist.dart';
+import 'package:over_react_analyzer_plugin/src/assist/toggle_stateful.dart';
+import 'package:test/test.dart';
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../test_bases/assist_test_base.dart';
+import 'boilerplate_assist_utils.dart';
+
+void main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(AddStatefulnessAssist);
+    defineReflectiveTests(RemoveStatefulnessAssist);
+  });
+}
+
+@reflectiveTest
+class AddStatefulnessAssist extends AssistTestBase with BoilerplateAssistTestStrings {
+  static int expectedPriority = ToggleComponentStatefulness.statefulAssistPriority;
+
+  @override
+  AssistKind get assistKindUnderTest => ToggleComponentStatefulness.makeStateful;
+
+  Future<void> test_noAssist() async {
+    final source = newSource('test.dart', 'var foo = true;');
+    final selection = createSelection(source, '#var foo = true;#');
+    await expectNoAssist(selection);
+  }
+
+  Future<void> test_noAssistOnFactory() async {
+    final generatedSource = simpleUiComponentSource();
+
+    var source = newSource(fileName, generatedSource);
+    var selection = createSelection(source, r'UiFactory<FooProps> #Foo# = _$Foo;');
+    await expectNoAssist(selection);
+  }
+
+  Future<void> test_noAssistOnProps() async {
+    final generatedSource = simpleUiComponentSource();
+
+    var source = newSource(fileName, generatedSource);
+    var selection = createSelection(source, 'mixin #FooProps# on UiProps {}');
+    await expectNoAssist(selection);
+  }
+
+  Future<void> test_noAssistOnOldBoilerplate() async {
+    const oldBoilerplate = '''
+import 'package:over_react/over_react.dart';
+part 'test.over_react.g.dart';
+
+@Factory()
+UiFactory<TestProps> Test = _\$Test;
+
+@Props()
+class _\$TestProps extends UiProps {}
+
+@Component()
+class TestComponent extends UiComponent<TestProps> {
+  @override
+  render() {}
+}
+''';
+
+    var source = newSource(fileName, oldBoilerplate);
+    var selection = createSelection(source, 'class #TestComponent# extends');
+    await expectNoAssist(selection);
+  }
+
+  Future<void> test_noAssistOnUnknownBase() async {
+    const oldBoilerplate = '''
```

Comment:
This should be the new boilerplate with an unknown base, otherwise it's not effectively exercising the logic here: https://github.com/Workiva/over_react/blob/master/tools/analyzer_plugin/lib/src/assist/toggle_stateful.dart#L49-L52

since it will have already short-circuited here: https://github.com/Workiva/over_react/blob/master/tools/analyzer_plugin/lib/src/assist/toggle_stateful.dart#L47